### DISCUSSION TITLE
useQuery same queryKey but different queryFn

### DISCUSSION THREAD

**Discussion by charshin** [TanStack/query#4020] [Q&A] [answered]
Let's say I have one useQuery, called `UQ1` with queryKey `QK` and queryFn `QF1`  in a mounted component, staleTime is set to 10s globally, `UQ1` performed fetching successfully, and data got cached with `QK`.

Before 10s elapses, I mount another component with useQuery `UQ2` with queryKey `QK` but queryFn `QF2`, I suppose it will get the cached data instead of performing another fetch with `QF2`, is that right?

After 10s, click away & refocus window, will both of the useQuery start a refetch or one of them only? I guess, both will be fired, and dedup will take place as in https://github.com/TanStack/query/discussions/608. Question is who will be picked first, it is not deterministic, isn't it? 

I am just asking for the sake of curiosity. Thanks all

**Comment by TkDodo** [comment, marked as answer]
>  I suppose it will get the cached data instead of performing another fetch with QF2, is that right?

yes that's right.

> Question is who will be picked first, it is not deterministic, isn't it?

"the first observer" will be picked, so the one that mounted first I guess. It's really not a good idea to do different Query Function for the same key!
